### PR TITLE
hyprlock: init at 0.1.0

### LINKED ISSUE #290334: Package request: `hyprlock`
**Project description**

Hyprland lock screen utility

**Metadata**

* homepage URL: https://wiki.hyprland.org/Hypr-Ecosystem/hyprlock/
* source URL: https://github.com/hyprwm/hyprlock/tree/main
* license: BSD3
* platforms: unix, linux, darwin, ...

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Fixes #290334.

Adds hyprlock, Hyprland's screen locker.

Homepage: https://github.com/hyprwm/hyprlock

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/hy/hyprlock/package.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+, hyprlang
+, pam
+, wayland
+, wayland-protocols
+, cairo
+, pango
+, unstableGitUpdater
+}:
+
+stdenv.mkDerivation rec {
```
Comment (fufexan) [inline pkgs/by-name/hy/hyprlock/package.nix:17]: `rec` doesn't look necessary here.

File: pkgs/by-name/hy/hyprlock/package.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  pname = "hyprlock";
+  version = "0-unstable-2024-02-20";
```
Comment (fufexan) [inline pkgs/by-name/hy/hyprlock/package.nix:19]: Is `0-` some new convention?

Comment (eclairevoyant) [reply pkgs/by-name/hy/hyprlock/package.nix:19]: <https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming>

Comment (fufexan) [reply pkgs/by-name/hy/hyprlock/package.nix:19]: So this is the successor of `0.pre+date=YYYY-MM-DD`. Got it.

**Review by IogaMaster — APPROVED** [review]

**Review by fufexan — APPROVED** [review]

**Review by IogaMaster — APPROVED** [review]
Looks great!

**Comment by Jappie3** [comment]
this should propbably be updated, Hyprlock v0.1.0 released 20 hours ago: https://github.com/hyprwm/hyprlock/releases/tag/v0.1.0

**Review by IogaMaster — APPROVED** [review]
This review has been done with [warden](https://github.com/iogamaster/warden), please report any issues!


<details><summary>Packages built</summary>
<p>

✅ hyprlock

### `nix-info -m` 
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05.20240216.5863c27`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.19.3`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`



</p>
</details>

File: pkgs/by-name/hy/hyprlock/package.nix [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    cairo
+    expat
+    fribidi
```
Comment (natsukium) [inline pkgs/by-name/hy/hyprlock/package.nix:50]: Do we really need newly added dependencies?

Comment (eclairevoyant) [reply pkgs/by-name/hy/hyprlock/package.nix:50]: I went through and added them as the build errored or threw warnings. AFAICT they are all required

**Review by jnsgruk — APPROVED** [review]

**Comment by natsukium** [comment]
@jnsgruk Could you please wait to merge until the discussion is resolved?
If the answer is clear and you do not feel the need to wait for a response, could you comment that you have confirmed that the concerns have been addressed?

**Comment by jnsgruk** [comment]
> @jnsgruk Could you please wait to merge until the discussion is resolved? If the answer is clear and you do not feel the need to wait for a response, could you comment that you have confirmed that the concerns have been addressed?

My apologies - I had this open from yesterday and totally missed your comment. I'll get a PR together to address your concerns.

**Comment by natsukium** [comment]
No worries, I often do the same thing. 😄

**Comment by jnsgruk** [comment]
> No worries, I often do the same thing. 😄

https://github.com/NixOS/nixpkgs/pull/291842 :slightly_smiling_face:

**Pull request merged**